src/dist/replication/common/fs_manager.h

```diff
@@ -90,19 +103,52 @@ class fs_manager
     bool for_each_dir_node(const std::function<bool(const dir_node &)> &func) const;
     void update_disk_stat();
 
+    // when executing "disk rebalance", we need find min and max node(sorted_by_density)
+    static bool sorted_by_density(const std::shared_ptr<dir_node> left,
+                                  const std::shared_ptr<dir_node> right)
```

Comment:
`disk_density` is not always greater than zero. For example, disk_ratio is 8%, 11%, total_disk_ratio is 10%, so the density will be -2 and 1. I suggest you to consider this condition, or guarantee `disk_density` greater than zero.

Reply:
Yeah, we just used it to judge whether it is balanced.

Reply:
I don't understand how can `disk_density` identify balance, could you please explain about it?

Reply:
It's actually the concept of `range` which used to calculate the `standard deviatio`, Negative means the disk is below average, Positive means above. Well, why you expect it should be positive?

Reply:
The comment `it's used to evaluate whether a disk is balanced` raise me a question that how this variety evaluate a disk is balanced or not. 
Just like my previous comment, for example, disk A ratio is 8%, disk B ratio is 11%, total_disk_ratio is 10%, so the density A will be -2, density B will be 1, how can -2 and 1 to evaluate A and B disk balance?
As a result, I thought you would like to use `density` to distinguish which disk is much more not-balanced, just like the example above, disk A is much more not-balanced, and its `density` is expected to be greater than B, but -2 < 1. So I suggest you to guarantee `density` be greater than zero.
However, the meaning of `density` is not what I thought, it can not be used to `evaluate whether a disk is balanced`,  your comment in code confused me.
I suggest you update your comment in code to explain what `disk_density` exactly used for.

Reply:
Yeah, good advice, the comment should be comprehensible to every one.